From 22772c87565e87e4fae135a14f2012614ce799bb Mon Sep 17 00:00:00 2001
From: Alexandre Torgue <alexandre.torgue@st.com>
Date: Tue, 3 Nov 2020 10:37:10 +0100
Subject: [PATCH] clocksource: stm32-lptimer: use wakeup capable instead of
 init wakeup

"wakeup-source" property describes a device which has wakeup capability but
should not force this device as a wakeup source.

Signed-off-by: Alexandre Torgue <alexandre.torgue@st.com>
Change-Id: Iee4fd2e735a69a7889d2f1a0a43d6bc93ca25b45
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/182927
Reviewed-by: CITOOLS <smet-aci-reviews@lists.codex.cro.st.com>
Reviewed-by: Fabrice GASNIER <fabrice.gasnier@st.com>
---
 drivers/clocksource/timer-stm32-lp.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

--- a/drivers/clocksource/timer-stm32-lp.c
+++ b/drivers/clocksource/timer-stm32-lp.c
@@ -168,9 +168,7 @@ static int stm32_clkevent_lp_probe(struc
 	}
 
 	if (of_property_read_bool(pdev->dev.parent->of_node, "wakeup-source")) {
-		ret = device_init_wakeup(&pdev->dev, true);
-		if (ret)
-			goto out_clk_disable;
+		device_set_wakeup_capable(&pdev->dev, true);
 
 		ret = dev_pm_set_wake_irq(&pdev->dev, irq);
 		if (ret)
